src/echo-app/src/renderer/src/plugin-settings-page.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useEffect, useState } from 'react'
+
+import {
+  ECHO_PLUGIN_CONFIG,
+  EchoPluginConfig,
+  EchoPluginConfigs,
+  EchoPluginHook
+} from 'echo-common'
+import * as path from 'path'
+import { importDevPlugin } from './dev-plugins'
+
+function filterPluginsByEnv(pluginConfigs: EchoPluginConfigs) {
+  return Object.values(pluginConfigs).filter(
+    (config) => (import.meta.env.MODE === 'development') === config.name.endsWith('-dev')
+  )
+}
+
+async function getPlugin(pluginConfig: EchoPluginConfig): Promise<EchoPluginHook> {
+  if (import.meta.env.MODE === 'development') {
+    const entry = await importDevPlugin(pluginConfig.name)
+    return entry.default()
+  } else {
+    const p = path.resolve(pluginConfig.path)
+    const entry = module.require(p)
+    const plugin: EchoPluginHook = entry()
+    return plugin
+  }
+}
+
+export const PluginSettingsPage: React.FC = () => {
+  useEffect(() => {
+    setAvailablePlugins(filterPluginsByEnv(ECHO_PLUGIN_CONFIG.loadPluginConfigs()))
+  }, [])
+
+  async function handleTogglePlugin(pluginConfig: EchoPluginConfig) {
+    const pluginConfigs = ECHO_PLUGIN_CONFIG.loadPluginConfigs()
+    const pluginEnabled = pluginConfigs[pluginConfig.name].enabled
+    const plugin = await getPlugin(pluginConfig)
+    if (!pluginEnabled) {
+      plugin.start()
+    } else {
+      plugin.destroy()
+    }
```

Comment:
I would make `await plugin.destroy()` to give the plugin the chacne to write data before it gets destroyed

Reply:
Okay I see. `destory` calls unregister Route. Within the call the plugin could write the `unregister` after Promise.resolve()
Do you delete the plugin afterwards? Then I would wait for the plugin, if not it would make no difference I guess?

Reply:
No, there isn't any deletion in this PR. I wouldn't delete anything except by "uninstalling" it, which isn't implemented in this PR